tests/pytests/unit/grains/test_core.py

```diff
@@ -1154,6 +1154,36 @@ def test_almalinux_8_os_grains():
     }
     _run_os_grains_tests(_os_release_data, {}, expectation)
 
+@pytest.mark.skip_unless_on_linux
+def test_virtuozzo_7_os_grains():
+    """
+    Test if OS grains are parsed correctly in Virtuozzo 7
+    """
+    _os_release_data = {
+        "NAME": "Virtuozzo",
+        "ID": "virtuozzo",
+        "PRETTY_NAME": "Virtuozzo release 7.5.4",
+        "VERSION": "7.5.4",
+        "ID_LIKE": "rhel fedora",
+        "VERSION_ID": "7",
+        "ANSI_COLOR": "0;31",
+        "CPE_NAME": "cpe:/o:virtuozzoproject:vz:7",
+        "HOME_URL": "http://www.virtuozzo.com",
+        "BUG_REPORT_URL": "https://bugs.openvz.org",
+    }
+    expectation = {
+        "os": "Virtuozzo",
+        "os_family": "RedHat",
+        "osfullname": "Virtuozzo",
+        "oscodename": "Virtuozzo release 7.5.4",
+        "osrelease": "7",
+        "osrelease_info": (7,),
```

Comment:
If the OS version is 7.5.4, I should be expecting the following:
```suggestion
        "oscodename": "Virtuozzo release 7.5.4",
        "osrelease": "7",
        "osrelease_info": (7, 5, 4),
```

Reply:
Understood. 

When we test this change the actual return is (7,). It's because osrelease is also simply `7`, a result of osrelease being detected from VERSION_ID, which does not contain point release information like other Red Hat based distros. We view this as a failing on their part with regards to compliance. 
```
[root@havz-dev ~]# salt-call grains.get oscodename
local:
    Virtuozzo release 7.5.4
[root@havz-dev ~]# salt-call grains.get osrelease_info
local:
    - 7
[root@havz-dev ~]# salt-call grains.get osrelease
local:
    7

```
I can agree this is not expected behavior for the grain.
Approaching this seems a little messy.

I attempted to remedy this pursuing modification of `_os_release_quirks_for_osrelease` in `grains/core.py`, however the code path for modifying `_os_release_to_grains` in `grains/core.py` is overwritten by this later path where [lsb_has_error](https://github.com/jealouscloud/salt/blob/29c7fcecd59751e9999f533bea6176e4586b1c45/salt/grains/core.py#L2154) comes back as True which leads the code path to `_legacy_linux_distribution_data`.

This seemed like the reasonable path for hacking in a fix for a sort of broken os-release file, but `_legacy_linux_distribution_data` overwrites my changes.

At this point I'm looking for a reasonable way to deal with this discrepancy. Any feedback is appreciated.

Reply:
If the breakage is on the os-release file, I'd say leave it as is. Salt parses it correctly.